src/fields/fp/mod.rs

```diff
@@ -727,7 +727,11 @@ impl<F: PrimeField> FieldVar<F, F> for FpVar<F> {
             (Constant(s), Constant(d)) => Ok(Constant(*s / *d)),
             (Var(s), Constant(d)) => Ok(Var(s.mul_constant(d.inverse().get()?))),
             (Constant(s), Var(d)) => Ok(Var(d.inverse()?.mul_constant(*s))),
-            (Var(s), Var(d)) => Ok(Var(d.inverse()?.mul(s))),
+            (Var(s), Var(d)) => Self::new_witness(self.cs(), || {
+                let denominator_inv_native = denominator.value()?.inverse().get()?;
```

Comment:
This prevents errors in when the output is not used, but the denominator is invalid. (eg in the scalar mul code)

```suggestion
                let d_inv = d.value()?.inverse().ok_or(F::zero())?;
```